tests/jlm/mlir/frontend/TestMlirToJlmConverter.cpp

```diff
@@ -450,11 +448,9 @@ TestCompZeroExt()
 
       // Traverse the rvsgd graph upwards to check connections
       std::cout << "Testing lambdaResultOriginNodeOuput\n";
-      jlm::rvsdg::node_output * lambdaResultOriginNodeOuput;
-      assert(
-          lambdaResultOriginNodeOuput = dynamic_cast<jlm::rvsdg::node_output *>(
-              convertedLambda->subregion()->result(0)->origin()));
-      Node * ZExtNode = lambdaResultOriginNodeOuput->node();
+      auto ZExtNode = jlm::rvsdg::TryGetOwnerNode<jlm::rvsdg::SimpleNode>(
+          *convertedLambda->subregion()->result(0)->origin());
+      assert(ZExtNode);
```

Comment:
The assert() is not really necessary here. The pointer is used further down.

Reply:
I think the code was asserting before (in slightly different ways), and I thought meaningful assert failure instead of default does not harm in any case.